mcm/json_layer/request.py

```diff
@@ -1872,9 +1872,46 @@ def get_stats(self, forced=False):
                                                   page=0,
                                                   limit=-1,
                                                   options={'key': prepid})
+        # easier to access this way below
+        stats_workflows_dict = dict([(stats_workflow.get('RequestName'), stats_workflow) for stats_workflow in stats_workflows])
+
         mcm_reqmgr_list = self.get_attribute('reqmgr_name')
         mcm_reqmgr_name_list = [x['name'] for x in mcm_reqmgr_list]
-        stats_reqmgr_name_list = [stats_wf['RequestName'] for stats_wf in stats_workflows]
+
+        # Attempt to pick the latest injected workflow using the records available in McM
+        # After an injection process finishes, only the latest workflow name is available on the `reqmgr_name`
+        # attribute with no content.
+        # Try to fetch it from Stats2 and pick the date from the `new` transition.
+        #
+        # Note this process is not idempotent. In case the request has been updated with Stats2
+        # data before, nothing will be filtered. Such behavior is acceptable.
+        workflows_to_consider = [
+            stats_workflows_dict.get(mcm_reqmgr_name)
+            for mcm_reqmgr_name in mcm_reqmgr_name_list
+            if stats_workflows_dict.get(mcm_reqmgr_name)
+        ]
+        self.logger.info("Workflows to consider: %s", [doc.get('RequestName') for doc in workflows_to_consider])
+        try:
+            reference_times_of_new = min([
```

Comment:
It's a single time, using a plural here is confusing:
```suggestion
            reference_time_of_new = min([
```